server/bin/test_data_importer.py

```diff
@@ -0,0 +1,868 @@
+            for product in data['products']:
+                product_map[product['id']] = product
+
+        if 'owners' in data:
+            for owner in data['owners']:
+                if 'products' in owner:
+                    for product in owner['products']:
+                        product_map[product['id']] = product
+
+        products.extend(order_products(None, product_map))
+    else:
+        # In standalone mode, we can only add stuff if we have an org to add it to
+        if 'owners' in data:
+            for owner in data['owners']:
+                owner_key = owner['name']
+                product_map = {}
+
+                if 'products' in data:
+                    for product in data['products']:
+                        product_map[product['id']] = product
+
+                if 'products' in owner:
+                    for product in owner['products']:
+                        product_map[product['id']] = product
+
+                products.extend(order_products(owner_key, product_map))
+
+    return products
+
+def map_content_to_product(cp, owner_key, product_data, content_data):
+    """Performs content generation and mapping for a given product"""
+
+    # TODO: FIXME: Redo the test_data.json format so content is defined as part of the product
+    # rather than screwing around with faking unique content and mapping it to each individual
+    # product. Content sharing is eventually going to be removed from Candlepin, and sharing
+    # a single content instance between multiple products breaks repo generation/management very
+    # badly.
+    def find_content(content_id):
+        if owner_key in content_data:
+            if content_id in content_data[owner_key]:
+                return content_data[owner_key][content_id]
+
+        global_content = content_data[None]
+
+        if content_id in global_content:
+            return global_content[content_id]
+
+        errmsg = 'product references a non-existent content: {name} ({pid}) => {ref_cid}'.format(
+            name=product_data['name'], pid=product_data['id'], ref_cid=content_id)
+
+        raise ReferenceError(errmsg)
+
+
+    content_map = {}
+    product_content = []
+
+    if 'content' in product_data:
+        # Test data stores this as an array of arrays for "reasons." Convert it to a dictionary and pass it
+        # to Candlepin
+        for content_id, enabled in product_data['content']:
+            # Create a unique content specifically for this product-content pairing.
+            content = find_content(content_id).copy()
+            content_id = '{pid}-{cid}'.format(pid=product_data['id'], cid=content_id)
+
+            # Modify the base content data so that it is unique for this product
+            content['id'] = content_id
+            content['name'] = '{name} ({uid})'.format(name=content['name'], uid=product_data['id'])
+            content['label'] = '{label} ({uid})'.format(label=content['label'], uid=product_data['id'])
```

Comment:
Nitpick. Can you please modify this line to this:

```python
content['label'] = '{label}-{uid}'.format(label=content['label'], uid=product_data['id'])
```

Without this change generated `repo_id` looks strange, because white spaces and special characters in content label are transformed to '-' by subscription-manager, when redha.repo is generated.

```
[root@thinkpad-t580 subscription-manager]# subscription-manager repos --list-enabled 
+----------------------------------------------------------+
    Available Repositories in /etc/yum.repos.d/redhat.repo
+----------------------------------------------------------+
Repo ID:   fake-content--38072-   <======== Too many '-'
Repo Name: fake-content (38072)
Repo URL:  http://svice:8080/path/to/fake-content/38072-3902
Enabled:   1

[root@thinkpad-t580 subscription-manager]# rct cc /etc/pki/entitlement/370058437736336375.pem | grep "Content:" -A 10
Content:
	Type: yum
	Name: fake-content (38072)
	Label: fake-content (38072)
	Vendor: Red Hat
	URL: /path/to/fake-content/38072-3902
	GPG: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-candlepin
	Enabled: True
	Expires: 
	Required Tags: 
	Arches: ALL
```

Reply:
Dnf/Yum accepts only alphanumeric characters and '-' in repository ID.

Reply:
Unfortunate. Alright, I'll remove the parens.

Reply:
After a few minutes of looking at this, I now realize that "Repo ID" is not the content ID, but either the name or the label of the content. Bah.